### ISSUE TITLE
fx:properties have to be within a service

### ISSUE THREAD

**Issue by justin2004** [SPARQL-Anything/sparql.anything#175]
similar to #173 

this yields results:
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
service <x-sparql-anything:>{
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
   graph ?g {?s ?p ?o }.
}
}'
```

whereas this does not yield results:
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
   graph ?g {?s ?p ?o }.
}'
```

 
I wonder if we should just make it mandatory to use a service clause so it simplifies the code?

**Comment by enridaga** [comment]
I think this is the current intended behaviour, where the SPARQL Anything engine only reacts to SERVICE clauses. In #150 we started to discuss the option of passing FX properties outside of the query, for example, as command line parameters or, in the case of the server version, as URL query string. The same applies to #173 unless I am missing the point entirely :)

**Comment by justin2004** [comment]
I think @luigi-asprino added the functionality that allows this (with no service clause) to work:
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
?s ?p ?o.
}'
```

**Comment by luigi-asprino** [comment]
Actually this is unintended. I'll mark this as bug. 
When analysing OpBGPs the FacadeXOpExecutor should check that the BGP is included in a service clause.

**Comment by luigi-asprino** [comment]
In a4df980 I disabled the possibility of passing parameters outside the service clause.
When processing OpBGP,  it seems that there is no way to figure out if the BGP is within a service or not. 
The work around implemented required to wrap the Jena's ExecutionContext within a FacadeXExecutionContext when processing the service clause. In this way, when the OpBGP is processed, the FacadeXOpExecutor can check whether it "receives" a  FacadeXExecutionContext or a ExecutionContext. @justin2004, does this solve the issue?

**Comment by justin2004** [comment]
@luigi-asprino 
i just checked and now this does not return results:
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
?s ?p ?o.
}'
```

but it did in December.

**Comment by justin2004** [comment]
i wonder if reacting to fx:properties outside of a service clause is more trouble than it is worth?

**Comment by enridaga** [comment]
At the moment, fx properties are not meant to work outside a service clause. However, we can still think of a no-service mode (see #150) where fx properties are supposed to be provided externally, for example, via CLI. A similar thing could be done via HTTP query parameters for the server mode.

**Comment by enridaga** [comment]
I just renamed the title for better clarity. I think this is solved and we can resume the discussion on #150
